api_types.ts

```diff
@@ -776,6 +777,11 @@ export interface ExecutionContext {
    * Information about state of the current sync. Only populated if this is a sync table formula.
    */
   readonly sync?: Sync;
+
+  /**
+   * {@link core.MetadataContext}
+   */
+  readonly parameters?: MetadataContext;
```

Comment:
If I remember correctly, when we originally discussed this you proposed putting it within the `Sync` object on execution context since it was going to hold the sync table parameters.

Is there an intent here to use the same field to support other, non-sync parameters on different types of formulas?

Reply:
yeah exactly, as per @patrick-codaio 's suggestion here: https://github.com/coda/packs-sdk/pull/2732#discussion_r1316207564

Reply:
What would we put in here for a non-sync formula? A duplicate version of the existing formula parameters indexed by parameter name strings instead of a list with positional indexes?

Reply:
I thought the general north star here would be to move this shared object into a standard place, instead of having it in separate different places for different formulas (so, when we eventually have an updated formula parameters shape that can handle the multiple set of varargs arguments, we could simply have all formulas pull from execution context instead of a positional array of arguments). 

Perhaps I am misunderstanding some of my earlier conversations with you and/or @patrick-codaio , or this PR is too ambitious?

Reply:
I'm not opposed to moving arguments from positional parameters to context for more types of arguments in more types of formulas, but my feeling is that sort of thing would be orthogonal to this PR which is only trying to put sync table parameters into the context.

`MetadataContext` is the only type we have for sync table parameters at the moment, so it seems like for now we're stuck with it. If we want to replace it with a new, incompatible type in the future we'd need to think about and design what that would look like.

Reply:
I'm fine with being less ambitious in this PR, I just thought we should be at least talking about the long term north star.

Reply:
Sounds good — moving it back into the `Sync` context for now. @dweitzman-codaio is that ok with you? I saw your original comment about putting it in a custom sync field for property options only, but I think we already get the parameters for normal sync formulas in `calc_service._getParameters`

Reply:
I'm fine with having the property defined in typescript for all sync table formulas. The only downside to putting it directly in Sync is that it's more work to ensure it's populated for all the formula types, but it seems like a manageable amount of work that's worth doing.